packages/components/src/components/button/button.tsx

```diff
@@ -80,13 +81,22 @@ export class Button {
   /** Add ARIA attributes. */
   @Prop() public aria?: SelectedAriaAttributes<ButtonAriaAttribute>;
 
+  private initialLoading: boolean = false;
+
   @Listen('click', { capture: true })
   public onClick(e: MouseEvent): void {
     if (isDisabledOrLoading(this.disabled, this.loading)) {
       e.stopPropagation();
     }
   }
 
+  @Watch('loading')
+  public loadingChanged(newVal: boolean): void {
```

Comment:
It would be worth looking into decorators to share this across components. Inheritance would also work but I guess isn't supported.
Also a generic test based on `component-meta` to establish this rule for all present and future components with the `loading` prop would be helpful.

Reply:
Do you mean abstracting the @Watch decorator by using it inside of a custom decorator which then can be called with smth. like this? `@CustomLoadingDecorator`